packages/shorebird_cli/lib/src/shorebird_process.dart

```diff
@@ -3,30 +3,34 @@ import 'dart:io';
 import 'package:meta/meta.dart';
 import 'package:shorebird_cli/src/shorebird_environment.dart';
 
-typedef RunProcess = Future<ProcessResult> Function(
-  String executable,
-  List<String> arguments, {
-  bool runInShell,
-  Map<String, String>? environment,
-  String? workingDirectory,
-  bool useVendedFlutter,
-});
-
-typedef StartProcess = Future<Process> Function(
-  String executable,
-  List<String> arguments, {
-  bool runInShell,
-  Map<String, String>? environment,
-  bool useVendedFlutter,
-});
+class EngineConfig {
+  const EngineConfig({
+    required this.localEngineSrcPath,
+    required this.localEngine,
+  });
+
+  const EngineConfig.empty()
```

Comment:
minor but couldn't we make the args in the default constructor optional and remove this extra constructor?

```dart
const EngineConfig({
  this.localEngineSrcPath,
  this.localEngine,
});
```